packages/core-data/src/utils/get-nested-value.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Helper util to return a value from a certain path of the object.
+ * Path is specified as either:
+ * - a string of properties, separated by dots, for example: "x.y".
+ * - an array of properties, for example `[ 'x', 'y' ]`.
+ * You can also specify a default value in case the result is nullish.
+ *
+ * @param {Object}       object       Input object.
+ * @param {string|Array} path         Path to the object property.
+ * @param {*}            defaultValue Default value if the value at the specified path is nullish.
+ * @return {*} Value of the object property at the specified path.
+ */
+export default function getNestedValue( object, path, defaultValue ) {
+	if (
+		! object ||
+		typeof object !== 'object' ||
+		( typeof path !== 'string' && ! Array.isArray( path ) )
+	) {
+		return object;
+	}
+	const normalizedPath = Array.isArray( path ) ? path : path.split( '.' );
+	let value = object;
+	normalizedPath.forEach( ( fieldName ) => {
+		value = value?.[ fieldName ];
+	} );
+	return value !== undefined ? value : defaultValue;
```

Comment:
The function and third argument description claim to use the `defaultValue` if the value at the specified path is nullish, however, this isn't true. It appears that it will be used only if the value is undefined. 

To fix that, you can use nullish coalescing instead, which is also shorter:

```suggestion
	return value ?? defaultValue;
```

Alternatively, you could use the `!= undefined` idiom, which also means "is not undefined and is not null":

```suggestion
	return value != undefined ? value : defaultValue;
```

however I believe this will raise an ESLint warning, so I'd suggest the first approach.

Reply:
> The function and third argument description claim to use the defaultValue if the value at the specified path is bullish

Good catch; I corrected the function jsdoc because that was the mistake. Returning null when the value is null is ok. The function will return `null` is the value is `null`, and the `defaultValue` if the value is `undefined`.

Reply:
Corrected the function doc: https://github.com/WordPress/gutenberg/pull/54161/commits/de832bf14a6789cdf0bec15c5117799b14622b7c